app/src/main/java/com/example/myapplication/SignUpDetailsActivity.kt

```diff
@@ -21,24 +25,65 @@ class SignUpDetailsActivity : AppCompatActivity() {
         val nextButton = findViewById<Button>(R.id.btnNext)
 
         nextButton.setOnClickListener {
-            val fullName = fullNameEditText.text.toString()
-            val validEmail = validEmailEditText.text.toString()
-            val phoneNumber = phoneNumberEditText.text.toString()
-            val strongPassword = strongPasswordEditText.text.toString()
+            val fullName = fullNameEditText.text.toString().trim()
+            val email = validEmailEditText.text.toString().trim()
+            val phone = phoneNumberEditText.text.toString().trim()
+            val password = strongPasswordEditText.text.toString().trim()
             val isTermsAccepted = termsAndConditionsCheckBox.isChecked
 
-            if (isTermsAccepted) {
-                Toast.makeText(this, "You have successfully registered!", Toast.LENGTH_SHORT).show()
-            } else {
-                Toast.makeText(this, "Please accept the Terms and Conditions", Toast.LENGTH_SHORT)
-                    .show()
+            when {
+                !credentialsManager.isValidFullName(fullName) -> {
```

Comment:
Same as above, how about making methods for displaying errors and only passing a string resource to it?